packages/wallet/src/PersonalWallet/PersonalWallet.ts

```diff
@@ -480,18 +494,20 @@ export class PersonalWallet implements ObservableWallet {
   }
 
   async submitTx(
-    input: Cardano.Tx | TxCBOR | OutgoingTx,
+    input: Cardano.Tx | TxCBOR | OutgoingTx | SubmitTxArgs,
     { mightBeAlreadySubmitted }: SubmitTxOptions = {}
   ): Promise<Cardano.TransactionId> {
     const outgoingTx = processOutgoingTx(input);
     this.#logger.debug(`Submitting transaction ${outgoingTx.id}`);
     this.#newTransactions.submitting$.next(outgoingTx);
     try {
       await this.txSubmitProvider.submitTx({
+        ...(outgoingTx.context && { context: { handles: outgoingTx.context } }),
         signedTransaction: outgoingTx.cbor
       });
       const { slot: submittedAt } = await firstValueFrom(this.tip$);
       this.#logger.debug(`Submitted transaction ${outgoingTx.id} at slot ${submittedAt}`);
+      this.#logger.debug(outgoingTx);
```

Comment:
Let's keep a single log per submission
```suggestion
      this.#logger.debug(`Submitted transaction ${outgoingTx.id} at slot ${submittedAt}; # handles: ${outgoingTx.context?.handles?.length || 0}`);
```

Reply:
2d88f9ab41c0bf941b8126409e8d597fa2713303  should address this